paddle/fluid/framework/details/multi_devices_graph_pass.cc

```diff
@@ -265,6 +265,16 @@ std::vector<ir::Node *> SortOpsAndDelayOptimizeOp(const ir::Graph &graph) {
       std::unordered_set<ir::Node *> optimize_set(optimize_ops.begin(),
                                                   optimize_ops.end());
       for (ir::Node *n : sorted_ret) {
+        // TODO(panyx0718): Revisit this soon.
+        if (boost::get<int>(
+                n->Op()->GetAttr(OpProtoAndCheckerMaker::OpRoleAttrName())) !=
+                static_cast<int>(OpRole::kBackward) &&
+            boost::get<int>(
```

Comment:
scale loss op should be skipped too? they are of role `kBackward | kLoss` or `kForward | kLoss`